From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Wed, 22 Jun 2022 02:21:03 +0300
Subject: [PATCH] Stackable items and rebranding


diff --git a/build.gradle.kts b/build.gradle.kts
index 68b8f275f9c9e77a949ca29749a05b469bcdb910..8fd0e3b0d060714b55d3891ec28d6b3b59710834 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -9,8 +9,12 @@ plugins {
 }
 
 dependencies {
-    implementation(project(":paper-api"))
-    implementation(project(":paper-mojangapi"))
+    // Whitehorse start
+    implementation(project(":whitehorse-api"))
+    implementation("io.papermc.paper:paper-mojangapi:1.19-R0.1-SNAPSHOT") {
+        exclude("io.papermc.paper", "paper-api")
+    }
+    // Whitehorse end
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
@@ -64,7 +68,7 @@ tasks.jar {
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
             "Implementation-Title" to "CraftBukkit",
-            "Implementation-Version" to "git-Paper-$implementationVersion",
+            "Implementation-Version" to "git-Whitehorse-$implementationVersion", // Whitehorse
             "Implementation-Vendor" to date, // Paper
             "Specification-Title" to "Bukkit",
             "Specification-Version" to project.version,
diff --git a/src/main/java/com/github/brominemc/whitehorse/WhitehorseVersionFetcher.java b/src/main/java/com/github/brominemc/whitehorse/WhitehorseVersionFetcher.java
new file mode 100644
index 0000000000000000000000000000000000000000..9e4fafa5e4df2c9629b691a0c70371c4608bfe30
--- /dev/null
+++ b/src/main/java/com/github/brominemc/whitehorse/WhitehorseVersionFetcher.java
@@ -0,0 +1,18 @@
+package com.github.brominemc.whitehorse;
+
+import com.destroystokyo.paper.util.VersionFetcher;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.jetbrains.annotations.NotNull;
+
+public class WhitehorseVersionFetcher implements VersionFetcher {
+    @Override
+    public long getCacheTime() {
+        return -1;
+    }
+
+    @Override
+    public @NotNull Component getVersionMessage(@NotNull String serverVersion) {
+        return Component.text("Whitehorse: ").append(Component.text(serverVersion, NamedTextColor.GREEN));
+    }
+}
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 77cd45f616e2ff38ad6a648b8b865a99e544f3ec..150195d699213a9a492bacaac5d1b22a59076e3c 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1654,7 +1654,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "Paper"; // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "Whitehorse"; // Whitehorse - Whitehorse > // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport details) {
diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index 6b3cfc19c4fd1382ddf534265a1114995a4f6b55..57a98c5628fbc1f41589ca37155c1ae7152cd193 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -514,6 +514,12 @@ public final class ItemStack {
     }
 
     public int getMaxStackSize() {
+        // Whitehorse start
+        if (this.tag != null && this.tag.contains("BromineMC.MaxStackSize")) {
+            byte maxStackSize = this.tag.getByte("BromineMC.MaxStackSize");
+            if (maxStackSize > 0) return maxStackSize;
+        }
+        // Whitehorse end
         return this.getItem().getMaxStackSize();
     }
 
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 61c2f864bb23e75f6c377f7ccbf2ec44a670348c..0988f768bddff85fdfe80872a20de0ecc3688f08 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -245,7 +245,7 @@ import javax.annotation.Nullable; // Paper
 import javax.annotation.Nonnull; // Paper
 
 public final class CraftServer implements Server {
-    private final String serverName = "Paper"; // Paper
+    private final String serverName = "Whitehorse"; // Whitehorse > // Paper
     private final String serverVersion;
     private final String bukkitVersion = Versioning.getBukkitVersion();
     private final Logger logger = Logger.getLogger("Minecraft");
diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
index f5363a753e5d24ccda946a9d65132eed28f19172..77cf10ff501f3bd7abc1605770e110902b8f96c5 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
@@ -170,7 +170,7 @@ public final class CraftItemStack extends ItemStack {
 
     @Override
     public int getMaxStackSize() {
-        return (this.handle == null) ? Material.AIR.getMaxStackSize() : this.handle.getItem().getMaxStackSize();
+        return (this.handle == null) ? Material.AIR.getMaxStackSize() : this.handle.getMaxStackSize(); // Whitehorse
     }
 
     // Paper start
diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
index 7c1e510a0ad4b69c1fedc3367a8216046efb228e..bb98fbbe8989e28d2651fa357566711a2ad00a75 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -430,7 +430,7 @@ public final class CraftMagicNumbers implements UnsafeValues {
 
     @Override
     public com.destroystokyo.paper.util.VersionFetcher getVersionFetcher() {
-        return new com.destroystokyo.paper.PaperVersionFetcher();
+        return new com.github.brominemc.whitehorse.WhitehorseVersionFetcher(); // Whitehorse
     }
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 774556a62eb240da42e84db4502e2ed43495be17..1d7839dbefc09491a2d018e176958a020b1e00db 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/io.papermc.paper/paper-api/pom.properties");
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/whitehorse/whitehorse-api/pom.properties"); // Whitehorse
         Properties properties = new Properties();
 
         if (stream != null) {
