From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Thu, 30 Mar 2023 09:34:56 +0300
Subject: [PATCH] Simple version fetcher


diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
index 0aee1fadaeb948fd90d8e61ac57dbe7f810529ff..3dcaf39a4b164643c6205e5425c42eded8014ab7 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -470,7 +470,7 @@ public final class CraftMagicNumbers implements UnsafeValues {
 
     @Override
     public com.destroystokyo.paper.util.VersionFetcher getVersionFetcher() {
-        return new com.destroystokyo.paper.PaperVersionFetcher(); // Pufferfish // Purpur
+        return new ru.brominemc.whitehorse.WhitehorseVersionFetcher(); // Pufferfish // Purpur // Whitehorse
     }
 
     @Override
diff --git a/src/main/java/ru/brominemc/whitehorse/WhitehorseVersionFetcher.java b/src/main/java/ru/brominemc/whitehorse/WhitehorseVersionFetcher.java
new file mode 100644
index 0000000000000000000000000000000000000000..7dd13fdd42832120556dabd3b8086b30ae509d9b
--- /dev/null
+++ b/src/main/java/ru/brominemc/whitehorse/WhitehorseVersionFetcher.java
@@ -0,0 +1,18 @@
+package ru.brominemc.whitehorse;
+
+import com.destroystokyo.paper.util.VersionFetcher;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.jetbrains.annotations.NotNull;
+
+public class WhitehorseVersionFetcher implements VersionFetcher {
+    @Override
+    public long getCacheTime() {
+        return Long.MAX_VALUE;
+    }
+
+    @Override
+    public @NotNull Component getVersionMessage(@NotNull String serverVersion) {
+        return Component.text("Whitehorse: ").append(Component.text(serverVersion, NamedTextColor.GREEN));
+    }
+}
