From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Tue, 7 Feb 2023 03:38:55 +0300
Subject: [PATCH] Stackable items


diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index 39a62d1dd85dbee349b05ac16ef4ee5572ddb1d6..08a7fca1e9ed6ad0841a67ea5a9c5ac286f7df62 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -526,6 +526,12 @@ public final class ItemStack {
     }
 
     public int getMaxStackSize() {
+        // Whitehorse start - stackable items
+        if (this.tag != null && this.tag.contains("BromineMC.MaxStackSize")) {
+            byte maxStackSize = this.tag.getByte("BromineMC.MaxStackSize");
+            if (maxStackSize > 0 && maxStackSize <= 64) return maxStackSize;
+        }
+        // Whitehorse end
         return this.getItem().getMaxStackSize();
     }
 
diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
index 97c5a72096ac0fd0b08a397466486014cfea4579..6fc82db64ac5d390605144cc043f7a4b140e5bb2 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
@@ -170,7 +170,7 @@ public final class CraftItemStack extends ItemStack {
 
     @Override
     public int getMaxStackSize() {
-        return (this.handle == null) ? Material.AIR.getMaxStackSize() : this.handle.getItem().getMaxStackSize();
+        return (this.handle == null) ? Material.AIR.getMaxStackSize() : this.handle.getMaxStackSize(); // Whitehorse - stackable items
     }
 
     // Paper start
