From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Sun, 28 Aug 2022 02:06:51 +0300
Subject: [PATCH] Remove chat validation


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index ea78fb81850168fd5bd63403c37caddafa075347..53ac956f72b89f423b7420753c26fca2c3e35796 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2339,6 +2339,8 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
     }
 
     private boolean tryHandleChat(String message, Instant timestamp, LastSeenMessages.Update acknowledgment) {
+        // Whitehorse start - remove chat validation
+        /*
         if (!this.updateChatOrder(timestamp)) {
             ServerGamePacketListenerImpl.LOGGER.warn("{} sent out-of-order chat: '{}': {} > {}", this.player.getName().getString(), message, this.lastChatTimeStamp.get().getEpochSecond(), timestamp.getEpochSecond()); // Paper
             this.server.scheduleOnMain(() -> { // Paper - push to main
@@ -2364,6 +2366,9 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                 return true;
             }
         }
+         */
+        return true;
+        // Whitehorse end
     }
 
     private boolean updateChatOrder(Instant timestamp) {
@@ -2595,6 +2600,8 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
     }
 
     private boolean verifyChatMessage(PlayerChatMessage playerchatmessage) {
+        // Whitehorse start - remove chat valdation
+        /*
         ChatSender chatsender = this.player.asChatSender();
         ProfilePublicKey profilepublickey = chatsender.profilePublicKey();
 
@@ -2614,6 +2621,8 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
             ServerGamePacketListenerImpl.LOGGER.warn("{} sent expired chat: '{}'. Is the client/server system time unsynchronized? c: {} s: {}", this.player.getName().getString(), playerchatmessage.signedContent().plain(), playerchatmessage.timeStamp().getEpochSecond(), Instant.now().getEpochSecond()); // Paper
         }
 
+        */
+        // Whitehorese end
         return true;
     }
 
