From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Thu, 30 Mar 2023 09:40:52 +0300
Subject: [PATCH] No chat reports


diff --git a/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java b/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java
index 74cf1c043beef03cfd5adf481414a5ee78bef2a6..823e17db76aaa4b2d18dd4137bf2555858559c9e 100644
--- a/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java
+++ b/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java
@@ -14,7 +14,7 @@ public interface OutgoingChatMessage {
     // Paper end
 
     static OutgoingChatMessage create(PlayerChatMessage message) {
-        return (OutgoingChatMessage)(message.isSystem() ? new OutgoingChatMessage.Disguised(message.decoratedContent()) : new OutgoingChatMessage.Player(message));
+        return new OutgoingChatMessage.Disguised(message.decoratedContent()); // Whitehorse - no chat reports
     }
 
     public static record Disguised(Component content) implements OutgoingChatMessage {
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 9c729910a39b010bf2123517c0bffc5a8993e6d1..1ea57f54c8cde8282fbaf8680052ee50d5cabacf 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -2153,7 +2153,7 @@ public class ServerPlayer extends Player {
     }
 
     public void sendServerStatus(ServerStatus metadata) {
-        this.connection.send(new ClientboundServerDataPacket(metadata.description(), metadata.favicon().map(ServerStatus.Favicon::iconBytes), metadata.enforcesSecureChat()));
+        this.connection.send(new ClientboundServerDataPacket(metadata.description(), metadata.favicon().map(ServerStatus.Favicon::iconBytes), true)); // Whitehorse - no chat reports
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 824f31b68b38f2f8642fb9d59a123cfdaffbb7b2..d47ceba390885d1032cf9b568295802cafd2bc45 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1479,7 +1479,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message) { // Paper - private -> public
-        return message.hasSignature() && !message.hasExpiredServer(Instant.now());
+        return true; // Whitehorse - no chat reports
     }
 
     // CraftBukkit start
