From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Tue, 7 Feb 2023 05:27:33 +0300
Subject: [PATCH] No chat reports


diff --git a/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java b/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java
index 74cf1c043beef03cfd5adf481414a5ee78bef2a6..823e17db76aaa4b2d18dd4137bf2555858559c9e 100644
--- a/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java
+++ b/src/main/java/net/minecraft/network/chat/OutgoingChatMessage.java
@@ -14,7 +14,7 @@ public interface OutgoingChatMessage {
     // Paper end
 
     static OutgoingChatMessage create(PlayerChatMessage message) {
-        return (OutgoingChatMessage)(message.isSystem() ? new OutgoingChatMessage.Disguised(message.decoratedContent()) : new OutgoingChatMessage.Player(message));
+        return new OutgoingChatMessage.Disguised(message.decoratedContent()); // Whitehorse - no chat reports
     }
 
     public static record Disguised(Component content) implements OutgoingChatMessage {
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 8664b5cfc739fc6b57643210cac3d03b279a59a1..7d1aa474d95f81c02a9bee3cba6885d03f7e6577 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -2073,7 +2073,7 @@ public class ServerPlayer extends Player {
     }
 
     public void sendServerStatus(ServerStatus metadata) {
-        this.connection.send(new ClientboundServerDataPacket(metadata.getDescription(), metadata.getFavicon(), metadata.enforcesSecureChat()));
+        this.connection.send(new ClientboundServerDataPacket(metadata.getDescription(), metadata.getFavicon(), true)); // Whitehorse - no chat reports
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 82093c413722e280b9fcb46fb0ba03ca629e54fa..8897460e1a6ccabbef16fb320071a86ae2896f94 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1450,7 +1450,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message) { // Paper - private -> public
-        return message.hasSignature() && !message.hasExpiredServer(Instant.now());
+        return true; // Whitehorse - no chat reports
     }
 
     // CraftBukkit start
