From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VidTu <60929459+VidTu@users.noreply.github.com>
Date: Sun, 5 Mar 2023 04:53:43 +0300
Subject: [PATCH] fixup! Stackable items


diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index 83e9ccb56b4725c3c29d33881d7c16bec5544b92..ec59f264bbe58b54b7a1136ef3307d3625eaed07 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -527,8 +527,8 @@ public final class ItemStack {
 
     public int getMaxStackSize() {
         // Whitehorse start - stackable items
-        if (this.tag != null && this.tag.contains("BromineMC.MaxStackSize")) {
-            byte maxStackSize = this.tag.getByte("BromineMC.MaxStackSize");
+        if (this.tag != null && this.tag.contains("Whitehorse.MaxStackSize")) {
+            byte maxStackSize = this.tag.getByte("Whitehorse.MaxStackSize");
             if (maxStackSize > 0 && maxStackSize <= 64) return maxStackSize;
         }
         // Whitehorse end
